File: sdks/ts/src/sdk/models/operations/putweblinksidnumberremovesharedlink.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PutWebLinksIdNumberRemoveSharedLinkPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=web_link_id" })
  webLinkId: string;
}


export class PutWebLinksIdNumberRemoveSharedLinkQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields: string;
}


export class PutWebLinksIdNumberRemoveSharedLinkRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=shared_link" })
  sharedLink?: Record<string, any>;
}


export class PutWebLinksIdNumberRemoveSharedLinkRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PutWebLinksIdNumberRemoveSharedLinkPathParams;

  @SpeakeasyMetadata()
  queryParams: PutWebLinksIdNumberRemoveSharedLinkQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PutWebLinksIdNumberRemoveSharedLinkRequestBody;
}


export class PutWebLinksIdNumberRemoveSharedLinkResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  webLink?: shared.WebLink;
}

File: sdks/go/pkg/models/operations/putweblinksidnumberremovesharedlink.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PutWebLinksIDNumberRemoveSharedLinkPathParams struct {
	WebLinkID string `pathParam:"style=simple,explode=false,name=web_link_id"`
}

type PutWebLinksIDNumberRemoveSharedLinkQueryParams struct {
	Fields string `queryParam:"style=form,explode=true,name=fields"`
}

type PutWebLinksIDNumberRemoveSharedLinkRequestBody struct {
	SharedLink map[string]interface{} `json:"shared_link,omitempty"`
}

type PutWebLinksIDNumberRemoveSharedLinkRequest struct {
	Retries     *utils.RetryConfig
	PathParams  PutWebLinksIDNumberRemoveSharedLinkPathParams
	QueryParams PutWebLinksIDNumberRemoveSharedLinkQueryParams
	Request     *PutWebLinksIDNumberRemoveSharedLinkRequestBody `request:"mediaType=application/json"`
}

type PutWebLinksIDNumberRemoveSharedLinkResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
	WebLink     *shared.WebLink
}
